vendor/blockchain_contracts/src/bin/calculate_offsets/ethereum/rfc003/contract.rs

```diff
@@ -0,0 +1,79 @@
+extern crate regex;
+use crate::calculate_offsets::ethereum::rfc003::{
+    compile_contract::compile, metadata::Metadata, offset::Offset,
+    placeholder_config::PlaceholderConfig, Error,
+};
+
+const TEMPLATE_FOLDER: &str = "./src/bin/calculate_offsets/ethereum/rfc003/templates/";
+const CONTRACT_FILE: &str = "contract.asm";
+const HEADER_FILE: &str = "deploy_header.asm";
+const CONFIG_FILE: &str = "config.json";
+
+pub struct Contract {
+    bytes: Vec<u8>,
+    placeholder_config: PlaceholderConfig,
+    pub meta_data: Metadata,
+}
+
+impl Contract {
+    pub fn compile_from_directory_and_load_placeholder_config(
+        dir: &'static str,
+    ) -> Result<Contract, Error> {
+        let mut bytes = compile(&get_header_file_path())?;
+        let mut contract_body = compile(&get_contract_file_path(dir))?;
+        bytes.append(&mut contract_body);
+
+        let placeholder_config = PlaceholderConfig::from_file(&get_config_file_path(dir))?;
+
+        let meta_data = Metadata::new(
+            placeholder_config.ledger_name.to_owned(),
+            placeholder_config.ledger_name.to_owned(),
+            hex::encode(bytes.to_owned()),
+        );
+
+        Ok(Self {
+            bytes,
+            placeholder_config,
+            meta_data,
+        })
+    }
+
+    pub fn calculate_offsets(&self) -> Result<Vec<Offset>, Error> {
+        let mut offsets: Vec<Offset> = Vec::new();
```

Comment:
:wrench: Instead of a mutable Vector and `push`ing into it, you could just to `into_iter` on a `clone()` if the placeholders and use `map` to turn it into a `Vec<Offset>`.

You might even want to return an `impl Iterator<Item=Offset>` if you want to avoid `collect()` right away.

Reply:
done, `collect()` is a cool thing :)